app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -84,7 +85,14 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
                             },
                             confirmCallback = { dialog ->
                                 viewModel.enableBiometry()
-                                viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                try {
+                                    //Some device still throw an IllegalStateException
+                                    // (due to no fingerprints enrolled) even though they return BIOMETRIC_SUCCESS
+                                    viewModel.encryptPasscodeWithBiometricKey(viewAction.passcode)
+                                } catch (e: Exception) {
+                                    settingsHandler.useBiometrics = false
```

Comment:
let's add disableBiometry to viewModel and set useBiometrics to false there, like it is set to true in enableBiometry()